openbas-api/src/main/java/io/openbas/rest/inject/service/ExecutableInjectService.java

```diff
@@ -0,0 +1,139 @@
+package io.openbas.rest.inject.service;
+
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import io.openbas.database.model.*;
+import io.openbas.database.repository.InjectRepository;
+import io.openbas.rest.exception.ElementNotFoundException;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class ExecutableInjectService {
+
+  private final InjectRepository injectRepository;
+  private static final Pattern argumentsRegex = Pattern.compile("#\\{([^#{}]+)}");
+
+  private List<String> getArgumentsFromCommandLines(String command) {
+    Matcher matcher = argumentsRegex.matcher(command);
+    List<String> commandParameters = new ArrayList<>();
+
+    while (matcher.find()) {
+      commandParameters.add(matcher.group(1));
+    }
+
+    return commandParameters;
+  }
+
+  private String replaceArgumentsByValue(
+      String command, List<PayloadArgument> defaultArguments, ObjectNode injectContent) {
+
+    List<String> arguments = getArgumentsFromCommandLines(command);
+
+    for (String argument : arguments) {
+      String value = "";
+
+      // Try to get the value from injectContent
+      if (injectContent.has(argument) && !injectContent.get(argument).asText().isEmpty()) {
+        value = injectContent.get(argument).asText();
+      } else {
+        // Fallback to defaultContent
+        value =
+            defaultArguments.stream()
+                .filter(a -> a.getKey().equals(argument))
+                .map(PayloadArgument::getDefaultValue)
+                .findFirst()
+                .orElse("");
+      }
+
+      command = command.replace("#{" + argument + "}", value);
+    }
+
+    return command;
+  }
+
+  private String processAndEncodeCommand(
+      String command,
+      String executor,
+      List<PayloadArgument> defaultArguments,
+      ObjectNode injectContent,
+      String obfuscator) {
+    String computedCommand = replaceArgumentsByValue(command, defaultArguments, injectContent);
+
+    if (executor.equals("cmd")) {
+      computedCommand = computedCommand.replace("\n", " & ");
```

Comment:
I think maybe you forgot the .trim() (it was in the implant code for cmd command)

Reply:
Thank you i didn't see that !